src/main/java/com/epam/labtaskspringcore/LabTaskSpringCoreApplication.java

```diff
@@ -1,133 +1,132 @@
-            log.error("could not create training");
-        } else {
-            Optional<Training> training4Optional = trainingService.getById(4);
-            if (training4Optional.isEmpty()) {
-                log.error("could not get training3");
-            } else {
-                System.out.println(training4Optional.get());
-            }
-        }
-        System.out.println("\tend of training 4 \n");
-
-        // updated trainee with id=2
-        System.out.println("\n \ttrainee id=2");
-        if (traineeService.getById(2).isEmpty()) {
-            log.error("No training with id=2 exists");
-        } else {
-            Trainee trainee2 = traineeService.getById(2).get();
-            trainee2.setLastName("Schmidt");
-            if (traineeService.update(trainee2).isEmpty()) {
-                log.error("could not update trainee2");
-            } else {
-                System.out.println(traineeService.getById(2).get());
-            }
-        }
-        System.out.println("\tend of trainee id=2 \n");
-    }
-}
+package com.epam.labtaskspringcore;
+
+import com.epam.labtaskspringcore.config.ApplicationContextProvider;
+import com.epam.labtaskspringcore.config.InMemoryStorage;
+import com.epam.labtaskspringcore.model.Trainee;
+import com.epam.labtaskspringcore.model.Trainer;
+import com.epam.labtaskspringcore.model.Training;
+import com.epam.labtaskspringcore.model.TrainingType;
+import com.epam.labtaskspringcore.service.TraineeService;
+import com.epam.labtaskspringcore.service.TrainerService;
+import com.epam.labtaskspringcore.service.TrainingService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+import java.util.Optional;
+
+@Slf4j
+@SpringBootApplication
+public class LabTaskSpringCoreApplication {
+    public static void main(String[] args) {
+
+        SpringApplication.run(LabTaskSpringCoreApplication.class, args);
+
+        InMemoryStorage inMemoryStorage = ApplicationContextProvider.getContext().getBean(InMemoryStorage.class);
+
+        log.info("\n\n>>>> START  ==============\n");
+        inMemoryStorage.getTrainees().values().stream().forEach(x -> log.info(x.toString()));
+        log.info("^^^^ TRAINEES\n");
+        inMemoryStorage.getTrainers().values().stream().forEach(x -> log.info(x.toString()));
+        log.info("^^^^ TRAINERS\n");
+        inMemoryStorage.getTrainings().values().stream().forEach(x -> log.info(x.toString()));
+        log.info("^^^^ TRAININGS\n");
+
+        TrainerService trainerService = ApplicationContextProvider.getContext().getBean(TrainerService.class);
+        TraineeService traineeService = ApplicationContextProvider.getContext().getBean(TraineeService.class);
+        TrainingService trainingService = ApplicationContextProvider.getContext().getBean(TrainingService.class);
+
+        log.info("\n \ttrainer 3 ");

```

Comment:
Could you please use a facade pattern for that https://refactoring.guru/design-patterns/facade/java/example or at lease split this logic to couple of private functions

Reply:
improved with commit cef2a5f